compiler/syntax/src/jsx_v4.ml

```diff
@@ -1208,7 +1208,7 @@ let append_children_prop (config : Jsx_common.jsx_config) mapper
           | "react" -> Lident "ReactDOM"
           | _generic -> module_access_name config "Elements"
         in
-        Exp.apply
+        Exp.apply ~loc:child.pexp_loc
```

Comment:
It looks like this change is responsible for the changes in the dead code analysis. But this change does not seem necessary to fix the location issue in error messages.

Reply:
I don't have an explanation as to why this would happen. More non-dummy locations should lead to more value dependencies found in the dead code analysis, which just traverses the typed ast.
Unless, somehow, this change affects the locations produced during type inference.